repository/src/main/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceFacadeFactoryBean.java

```diff
@@ -351,59 +360,106 @@ else if (isDefined(config.getTruststore()))
         {
             return possibleMetadataURIs()
-                    .stream()
-                    .map(u -> extractMetadata(rest, u))
-                    .filter(Optional::isPresent)
-                    .map(Optional::get)
-                    .findFirst()
-                    .map(this::createBuilder)
-                    .map(this::configureClientAuthentication)
-                    .map(Builder::build)
-                    .orElseThrow(() -> new IllegalStateException("Failed to create ClientRegistration."));
+                .stream()
+                .map(u -> extractMetadata(rest, u))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .findFirst()
+                .map(this::validateDiscoveryDocument)
+                .map(this::createBuilder)
+                .map(this::configureClientAuthentication)
+                .map(Builder::build)
+                .orElseThrow(() -> new IllegalStateException("Failed to create ClientRegistration."));
+        }
+
+        private OIDCProviderMetadata validateDiscoveryDocument(OIDCProviderMetadata metadata)
+        {
+            validateOIDCEndpoint(metadata.getTokenEndpointURI(), "Token");
+            validateOIDCEndpoint(metadata.getAuthorizationEndpointURI(), "Authorization");
+            validateOIDCEndpoint(metadata.getUserInfoEndpointURI(), "User Info");
+            validateOIDCEndpoint(metadata.getJWKSetURI(), "JWK Set");
+
+            if (metadata.getIssuer() != null)
+            {
+                try
+                {
+                    URI metadataIssuerURI = new URI(metadata.getIssuer().getValue());
+                    validateOIDCEndpoint(metadataIssuerURI, "Issuer");
+                    if (StringUtils.isNotBlank(config.getIssuerUrl()) &&
+                        !metadataIssuerURI.equals(URI.create(config.getIssuerUrl())))
+                    {
+                        throw new IdentityServiceException("Failed to create ClientRegistration. "
+                            + "The Issuer value from the OIDC Discovery does not align with the provided Issuer. Expected `%s` but found `%s`"
+                            .formatted(config.getIssuerUrl(), metadata.getIssuer().getValue()));
+                    }
+                }
+                catch (URISyntaxException e)
+                {
+                    throw new IdentityServiceException("The Issuer value could not be parsed as a URI reference.", e);
+                }
+            }
+            else
+            {
+                throw new IdentityServiceException("The Issuer cannot be null.");
+            }
+
+            return metadata;
+        }
+
+        private void validateOIDCEndpoint(URI value, String endpointName)
+        {
+            if (value == null || value.toASCIIString().isBlank())
+            {
+                throw new IdentityServiceException("The `%s` Endpoint cannot be empty.".formatted(endpointName));

```

Comment:
Please update the exception message to more clearly indicate that the value retrieved from the discovery endpoint document is invalid.

Reply:
Done.